sign-compare warning in Marshal.hpp

Upgraded from Ubuntu18.04 to Ubuntu20.04 and am now getting a (new) `sign-compare` warning in Marshall.hpp:
```
djinni/jni/Marshal.hpp: In static member function ‘static djinni::LocalRef<_jbyteArray*> djinni::Binary::fromCpp(JNIEnv*, const CppType&)’:
djinni/jni/Marshal.hpp:246:29: warning: comparison of integer expressions of different signedness: ‘std::vector<unsigned char>::size_type’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
  246 |             assert(c.size() <= std::numeric_limits<jsize>::max());
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Most likely due to and upgrade of clang.  I'm going to try to fix this in a fork and issue a PR.  I'll also try to update w/ compiler details.

--- a/djinni/jni/Marshal.hpp
+++ b/djinni/jni/Marshal.hpp
@@ -243,7 +243,15 @@ namespace djinni
 
         static LocalRef<JniType> fromCpp(JNIEnv* jniEnv, const CppType& c)
         {
-            assert(c.size() <= std::numeric_limits<jsize>::max());
+            // according to https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/types.html
+            //    `typedef jint jsize;`
+            // and
+            //    Java Type    Native Type     Description
+            //    int          jint            signed 32 bits
+            // therefore, casting numeric_limits<jsize>::max() to uint32_t
+            // would be valid and result in the same value, but now an
+            // unsigned type which can be compared to c.size()
+            assert(c.size() <= static_cast<uint32_t>(std::numeric_limits<jsize>::max()));
             auto j = LocalRef<jbyteArray>(jniEnv, jniEnv->NewByteArray(static_cast<jsize>(c.size())));
             jniExceptionCheck(jniEnv);
             // Using .data() on an empty vector is UB
@@ -371,7 +379,13 @@ namespace djinni
         static LocalRef<JniType> fromCpp(JNIEnv* jniEnv, const CppType& c)
         {
             const auto& data = JniClass<ListJniInfo>::get();
-            assert(c.size() <= std::numeric_limits<jint>::max());
+            // according to https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/types.html
+            //    Java Type    Native Type     Description
+            //    int          jint            signed 32 bits
+            // therefore, casting numeric_limits<jint>::max() to uint32_t
+            // would be valid and result in the same value, but now an
+            // unsigned type which can be compared to c.size()
+            assert(c.size() <= static_cast<uint32_t>(std::numeric_limits<jint>::max()));
             auto size = static_cast<jint>(c.size());
             auto j = LocalRef<jobject>(jniEnv, jniEnv->NewObject(data.clazz.get(), data.constructor, size));
             jniExceptionCheck(jniEnv);

--- a/djinni/jni/Marshal.hpp
+++ b/djinni/jni/Marshal.hpp
@@ -243,14 +243,6 @@ namespace djinni
 
         static LocalRef<JniType> fromCpp(JNIEnv* jniEnv, const CppType& c)
         {
-            // according to https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/types.html
-            //    `typedef jint jsize;`
-            // and
-            //    Java Type    Native Type     Description
-            //    int          jint            signed 32 bits
-            // therefore, casting numeric_limits<jsize>::max() to uint32_t
-            // would be valid and result in the same value, but now an
-            // unsigned type which can be compared to c.size()
             assert(c.size() <= static_cast<uint32_t>(std::numeric_limits<jsize>::max()));
             auto j = LocalRef<jbyteArray>(jniEnv, jniEnv->NewByteArray(static_cast<jsize>(c.size())));
             jniExceptionCheck(jniEnv);
@@ -379,12 +371,6 @@ namespace djinni
         static LocalRef<JniType> fromCpp(JNIEnv* jniEnv, const CppType& c)
         {
             const auto& data = JniClass<ListJniInfo>::get();
-            // according to https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/types.html
-            //    Java Type    Native Type     Description
-            //    int          jint            signed 32 bits
-            // therefore, casting numeric_limits<jint>::max() to uint32_t
-            // would be valid and result in the same value, but now an
-            // unsigned type which can be compared to c.size()
             assert(c.size() <= static_cast<uint32_t>(std::numeric_limits<jint>::max()));
             auto size = static_cast<jint>(c.size());
             auto j = LocalRef<jobject>(jniEnv, jniEnv->NewObject(data.clazz.get(), data.constructor, size));

